### ISSUE TITLE
Doesn't work with server side rendering (ReactRouterSSR+Kuronin:Pagination)

### ISSUE THREAD

**Issue by jsiig** [Kurounin/react-bootstrap-pagination#2]
What could be the issue?

[Meteor React Router SSR](https://github.com/thereactivestack/meteor-react-router-ssr)

Getting the following server console error and server side rendering fails, when using react-bootstrap-pagination along with Kuronin:Pagination
`
W20161021-15:15:26.072(3)? (STDERR) Fri Oct 21 2016 15:15:26 GMT+0300 (EEST) 'error while server-rendering' 'TypeError: props.pagination.ready is not a function\n    at [object Object].setDisplayedPages (/Users/johannes/Projects/TotoBet/totobet/node_modules/react-bootstrap-pagination/dist/pagination.js:62:35)\n    at [object Object].componentWillMount (/Users/johannes/Projects/TotoBet/totobet/node_modules/react-bootstrap-pagination/dist/pagination.js:46:18)\n    at /Users/johannes/Projects/TotoBet/totobet/node_modules/react/lib/ReactCompositeComponent.js:347:23\n    at measureLifeCyclePerf (/Users/johannes/Projects/TotoBet/totobet/node_modules/react/lib/ReactCompositeComponent.js:74:12)\n    at [object Object].ReactCompositeComponentMixin.performInitialMount (/Users/johannes/Projects/TotoBet/totobet/node_modules/react/lib/ReactCompositeComponent.js:346:9)\n    at [object Object].ReactCompositeComponentMixin.mountComponent (/Users/johannes/Projects/TotoBet/totobet/node_modules/react/lib/ReactCompositeComponent.js:257:21)\n    at Object.ReactReconciler.mountComponent (/Users/johannes/Projects/TotoBet/totobet/node_modules/react/lib/ReactReconciler.js:47:35)\n    at ReactDOMComponent.ReactMultiChild.Mixin.mountChildren (/Users/johannes/Projects/TotoBet/totobet/node_modules/react/lib/ReactMultiChild.js:240:44)\n    at ReactDOMComponent.Mixin._createContentMarkup (/Users/johannes/Projects/TotoBet/totobet/node_modules/react/lib/ReactDOMComponent.js:661:32)\n    at ReactDOMComponent.Mixin.mountComponent (/Users/johannes/Projects/TotoBet/totobet/node_modules/react/lib/ReactDOMComponent.js:528:29)'
`

Everything is fine if I disable SSR, but once I enable it it doesn't pre-render.

**Comment by Kurounin** [comment]
I created a repository with a react example: https://github.com/Kurounin/PaginationReactExample can you please submit a PR on that repository which replicates the problem?

**Comment by jsiig** [comment]
Just move main.jsx and main.html to the meteor root folder to enable SSR (in your repo). You should see exactly the same error.

Using your repo: 

```

I20161026-15:14:20.110(3)? Ignoring duplicate publish named 'myCollectionName'
W20161026-15:14:20.110(3)? (STDERR) Wed Oct 26 2016 15:14:20 GMT+0300 (EEST) 'error while server-rendering' 'TypeError: params.pag.getPage is not a function\n    at main.jsx:34:27\n    at [object Object].getMeteorData (packages/react-meteor-data/createContainer.jsx:33:14)\n    at MeteorDataManager.calculateData (packages/react-meteor-data/ReactMeteorData.jsx:61:24)\n    at [object Object].componentWillMount (packages/react-meteor-data/ReactMeteorData.jsx:5:45)\n    at /Users/johannes/Projects/PaginationReactExample/node_modules/react/lib/ReactCompositeComponent.js:347:23\n    at measureLifeCyclePerf (/Users/johannes/Projects/PaginationReactExample/node_modules/react/lib/ReactCompositeComponent.js:74:12)\n    at [object Object].ReactCompositeComponentMixin.performInitialMount (/Users/johannes/Projects/PaginationReactExample/node_modules/react/lib/ReactCompositeComponent.js:346:9)\n    at [object Object].ReactCompositeComponentMixin.mountComponent (/Users/johannes/Projects/PaginationReactExample/node_modules/react/lib/ReactCompositeComponent.js:257:21)\n    at Object.ReactReconciler.mountComponent (/Users/johannes/Projects/PaginationReactExample/node_modules/react/lib/ReactReconciler.js:47:35)\n    at [object Object].ReactCompositeComponentMixin.performInitialMount (/Users/johannes/Projects/PaginationReactExample/node_modules/react/lib/ReactCompositeComponent.js:370:34)'
```

**Comment by Kurounin** [comment]
The problem seems to be that the SSR expects all packages to publish the methods on the server side as well. Even if I modify my package, I still have a dependency on tmeasday:publish-counts package which doesn't publish the Counts on the server side (only on client side).
The solution would involve getting rid of the dependency and do everything in this package to be able to run the same methods on server side as well, but I don't think it would be worth the effort.

What is your main reason to try and use SSR, because you will most likely encounter problems with other packages as well?

**Comment by jsiig** [comment]
I actually just figured out how to solve it - I rewrote the package to publish Counts on the server side as well, and then renamed your server side pagination.js's Meteor.Pagination. I'll submit relevant PRs or fork the repositories when I can (soon).

My main reason for using SSR is that it needs to be an SPA with some public-facing parts and those definitely need to be curlable (and need SEO for search engines other than Google as well), but the app itself needs to also work reactively.
